### DISCUSSION TITLE
Dashboard and Monitoring for Self Hosted Option?

### DISCUSSION THREAD

**Discussion by niklasgrewe** [directus/directus#2812] [Archive]
Hello, everyone,
first of all I would like to thank you for this great project. Directus is really excellent and flexible solution in the CMS area. I now see that it is even possible to integrate dashboards. But are the dashboards only for the cloud version? I didn't find a way to create dashboards in my self-hosted instance, only collections. 

I can understand that some features are only available in the cloud, but does that have to apply to dashboards as well? For me personally, it would be much easier if I could display my website analytics in Directus without having to use the cloud variant. 

Are there any possibilities to implement this? Please correct my way of thinking here. I am just starting to get to know Directus 😉

**Comment by rijkvanzanten** [comment]
Heya! There are no "native dashboards" in Directus yet, but you should be able to whip something up pretty fast by using a custom module combined with a graph library like Chart.js 🙂

**Reply by niklasgrewe** [reply]
ok that means I could create such a dashboard as shown in the picture with a custom model and Chart.js myself, right? Are there any instructions or documentation so I can get an idea of how to create something like that for Directus?

![50868560-1a538900-137e-11e9-8ce9-83898b3b1858](https://user-images.githubusercontent.com/61348836/89433034-1cd8b900-d742-11ea-8e26-a6c3f87d9378.png)

Do you plan to support the native soon, maybe in v9? I can imagine that this is certainly a feature that is often requested, especially for website statistics. So you have everything in one place.

**Reply by rijkvanzanten** [reply]
That screenshot was just [a design exploration](https://github.com/directus/app/issues/1236#issue-397159231) for what such a feature could look like. Here's the docs on how to setup a custom module: https://docs.directus.io/extensions/modules.html

**Reply by benhaynes** [reply]
To add to this, we do plan on having a general charting dashboard in version 9... but it will be a one size fits all solution for general data. Version 9 will have a slightly different way to build these module extensions, and that information will be updated in the docs as we approach the official production release of version 9.

**Reply by niklasgrewe** [reply]
@benhaynes thank you for the helpful information, I'm glad to hear it. Can you tell me when v9 will come out?  I don't think I've seen anything on your roadmap yet...

**Reply by benhaynes** [reply]
There's an alpha out now, but there's no official release date for production. Likely one to two months. Although versions before then will likely be usable, especially for developing extensions.

**Reply by rijkvanzanten** [reply]
See https://github.com/directus/next 🙂

**Reply by sercanvirlan** [reply]
I tutorial would be awesome about how to manage navigation, sidebar maybe a boilerplate can hit this needs.  The design above looks gorgeous!!
